pkg/synthesis/synthesis_test.go

```diff
@@ -102,208 +120,249 @@ var groupsByVmsTests = []synthesisTest{
+}
+
+// ///////////////////////////////////////////////////////////////////////////////////////////////////////
+// serialTestsRun() gets a resource, and run the test functions serially
+// we need it to be serially, because we have only one resource
+func serialTestsRun(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel) {
+	runPreprocessing(synTest, t, rc)
+	runConvertToAbstract(synTest, t, rc)
+	runK8SSynthesis(synTest, t, rc)
+	runCompareNSXConnectivity(synTest, t, rc)
+}
+
+// parallelTestsRun() gets a test function to run, and run it on all the syntheticTests in parallel
+func parallelTestsRun(t *testing.T, f func(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel)) {
+	logging.Init(logging.HighVerbosity)
+	for _, test := range allSyntheticTests {
+		rc := data.ExamplesGeneration(&test.exData.FromNSX)
+		t.Run(test.name, func(t *testing.T) {
+			f(&test, t, rc)
+		},
+		)
 	}
-	baseName := fmt.Sprintf("%s_%t_%d", synTest.name, synTest.noHint, synTest.allowOnlyFromCategory)
-	outDir := path.Join("out", baseName)
-	fmt.Println("suffix:", suffix)
-	abstractModel, err := NSXToK8sSynthesis(rc, outDir, hintsParm, synTest.allowOnlyFromCategory)
-	expectedOutputFileName := filepath.Join(getTestsDirOut(), synTest.name+suffix)
-	expectedOutputDir := filepath.Join(getTestsDirOut(), k8sResourcesDir, baseName)
-	require.Nil(t, err)
-	addDebugFiles(t, rc, abstractModel, outDir)
-	actualOutput := strAllowOnlyPolicy(abstractModel.policy[0])
-	fmt.Println(actualOutput)
-	compareOrRegenerateOutputPerTest(t, mode, actualOutput, expectedOutputFileName, synTest.name)
-	compareOrRegenerateOutputDirPerTest(t, mode, filepath.Join(outDir, k8sResourcesDir), expectedOutputDir, synTest.name)
 }
 
-func addDebugFiles(t *testing.T, rc *collector.ResourcesContainerModel, abstractModel *AbstractModelSyn, outDir string) {
-	connectivity := map[string]string{}
-	var err error
-	// generate connectivity analysis output from the original NSX resources
-	debugDir := path.Join(outDir, "debug_resources")
+//////////////////////////////////////////
+// the test functions:
+//////////////////////////////////////////
 
-	for _, format := range []string{"txt", "dot"} {
-		params := common.OutputParameters{
-			Format: format,
-		}
-		connectivity[format], err = model.NSXConnectivityFromResourcesContainer(rc, params)
-		require.Nil(t, err)
-		err = common.WriteToFile(path.Join(debugDir, "vmware_connectivity."+format), connectivity[format])
-		require.Nil(t, err)
+func runPreprocessing(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel) {
+	// get the config:
+	parser := model.NewNSXConfigParserFromResourcesContainer(rc)
+	err := parser.RunParser()
+	require.Nil(t, err)
+	config := parser.GetConfig()
+	// write the config summery into a file, for debugging:
+	configStr := config.GetConfigInfoStr(false)
+	err = common.WriteToFile(path.Join(synTest.debugDir(), "config.txt"), configStr)
```

Comment:
is there a different debug dir per each test kind (abstract / pre-processing/ k8s-synth)?
or is it common for those tests and gets overridden by that last test kind that runs?

Reply:
there is a debug dir per synthesisTest.
for convenient, all files a in one dir